### PR TITLE
Feat : Install MATLAB on OOD

### PR DESCRIPTION
**Pull request by krishmoodbidri**

### CODE REVIEW

File: roles/ohpc_matlab/tasks/main.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+- name: install the latest version of libXtst
+  yum:
+    name: libXtst
+    state: latest
+
+- name: Download matlab
+  get_url:
+    url: https://uab.box.com/v/matlab
```
Comment (jprorama) [inline roles/ohpc_matlab/tasks/main.yaml]: How was this URL generated?

File: ohpc.yaml [hunk 2]
```diff
@@ -7,3 +7,4 @@
     - { name: 'compute_build_vnfs', tags: 'compute_build_vnfs' }
     - { name: 'compute_build_nodes', tags: 'compute_build_nodes' }
     - { name: 'nodes_vivify', tags: 'nodes_vivify' }
+    - { name: 'ohpc_matlab', tags: 'ohpc_matlab' }
```
Comment (jprorama) [inline ohpc.yaml]: Is there a way to make such components optional?  Either by setting a flag in groups_var/all or an explicit run of this role when matlab support is desired?  What would be the recommended approaches to accomplish this?

File: roles/ood/tasks/main.yaml [hunk 3]
```diff
@@ -195,4 +195,3 @@
     name: slurmd
     state: started
     enabled: yes
-
```
Comment (jprorama) [inline roles/ood/tasks/main.yaml]: Stray delete?

File: ood.yaml [hunk 4]
```diff
@@ -2,3 +2,4 @@
 - hosts: ood
   roles:
     - ood
+    - { name: 'oodmatlab', tags: 'oodmatlab' }
```
Comment (jprorama) [inline ood.yaml]: Let's use a similar naming convention.  If we are calling something ohpc_matlab above then this would more naturally be ood_matlab.

File: roles/ohpc_matlab/tasks/main.yaml (hunk 1 above)
Comment (krishmoodbidri) [reply roles/ohpc_matlab/tasks/main.yaml]: Custom URL shared link settings option on box

Comment (jprorama) [reply roles/ohpc_matlab/tasks/main.yaml]: Let's at least make this a variable in groups_var/all  so it can reasonably modified outside the playbook.

File: roles/oodmatlab/tasks/main.yaml [hunk 5]
```diff
@@ -0,0 +1,15 @@
+- name: matlab directory
```
Comment (jprorama) [inline roles/oodmatlab/tasks/main.yaml]: In changing the name of the role form oodmatlab, did you use git mv or just move files and add new ones?  It's better to rename files with git so that the history of edits are maintained.

As such, it appears that you still have two distinct role names in your repo, "ood_matlab" and "oodmatlab".

File: roles/ood_matlab/tasks/main.yaml [hunk 6]
```diff
@@ -8,7 +8,7 @@
     dest: /var/www/ood/apps/sys/matlab/
     clone: yes
 
-name: Replace submit.yml.erb
+- name: Replace submit.yml.erb
```
Comment (jprorama) [inline roles/ood_matlab/tasks/main.yaml]: Just a note, avoid adding unrelated changes to a commit.  The summary for the commit is about the httpd task so the changes should only relate to that task.  Make a correction like a missing dash a separate commit.  This makes it easier to track changes and cherry pick if needed.

File: group_vars/all [hunk 7]
```diff
@@ -109,6 +109,17 @@
   easybuild_buildpath: "/tmp/build"
   easybuild_sourcepath: "/tmp/source"
 
+#Slurm 17.11.11 patch
```
Comment (jprorama) [inline group_vars/all]: The slurm edits shouldn't be part of your branch.  They are already on uab-dev.

File: group_vars/all [hunk 8]
```diff
@@ -109,6 +109,17 @@
...
+  slurm_update: true
```
Comment (jprorama) [inline group_vars/all]: Again, slurm edits should not be part of this branch.

File: ohpc.yaml [hunk 9]
```diff
@@ -8,4 +8,5 @@
     - { name: 'compute_build_nodes', tags: 'compute_build_nodes' }
     - { name: 'nodes_vivify', tags: 'nodes_vivify' }
     - { name: 'ohpc_add_easybuild', tags: 'ohpc_add_easybuild' }
+    - { name: 'ohpc_matlab', when: matlab_provision, tags: 'ohpc_matlab' }
```
Comment (jprorama) [inline ohpc.yaml]: Could you change the order of these terms (name, tags, when).

File: roles/ohpc_matlab/tasks/main.yaml [hunk 10]
```diff
@@ -0,0 +1,12 @@
...
+    name: libXt
+    state: latest
+
+- name: Download matlab
+  get_url:
+    url: "{{ matlab_download_url }}"
+    dest: "{{ matlab_destination  }}"
+
+- name: Un-tar downloaded MATLAB 
+  command: tar -xvf {{ matlab_destination  }} -C "{{ matlab_clustershare }}"
```
Comment (jprorama) [inline roles/ohpc_matlab/tasks/main.yaml]: I'm getting an error on this task because the "-C" tries to change to a directory that doesn't exist.  You may need a task that creates the path first.

Also remove the "v" from the tar command.  There is no need to enumerate the file listing during extraction.

I also got a warning to consider using the unarchive module.  Could this be a better solution here?

**Comment by jprorama** [comment]
I'm getting an error when I try to launch the matlab interactive application.  The output.log file in the job directory shows:
```
No modules loaded + matlab -desktop /home/vagrant/ondemand/data/sys/dashboard/batch_connect/sys/matlab/output/29620884-b0a7-4375-be48-b2b13ef7d3e8/script.sh: line 37: matlab: command not found Cleaning up... Killing Xvnc process ID 3218 xfwm4: Fatal IO error 2 (No such file or directory) on X server :1.
```

It seems the user environment for the matlab executable isn't getting set up right.

File: roles/ohpc_matlab/tasks/main.yaml (hunk 10 above)
Comment (krishmoodbidri) [reply roles/ohpc_matlab/tasks/main.yaml]: I believe the -C in tar creates the directory if not found (http://man.openbsd.org/tar)
Removing v in the tar command

Comment (jprorama) [reply roles/ohpc_matlab/tasks/main.yaml]: That man page doesn't explicitly say it will be created, just that it is extract to that directory.  Also, we are on linux not openbsd so take care to ensure feature parity.  BSD and FSF versions of core tools sometimes have slightly different behavior in these edge cases.

Comment (jprorama) [reply roles/ohpc_matlab/tasks/main.yaml]: Also you might try the unarchive module and see if that works better than an explicit tar command
